sci-geosciences/mapserver/mapserver-7.6.0.ebuild

```diff
@@ -0,0 +1,286 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# NOTE: We have to stick with EAPI 6 for now because of the
+# depend.apache eclass.
+EAPI=6
+
+# Variables for the miscellaneous bindings we provide
+PHP_EXT_OPTIONAL_USE="php"
+PHP_EXT_NAME="php_mapscriptng"
+PHP_EXT_SKIP_PHPIZE="yes"
+
+USE_PHP="php7-2 php7-3 php7-4"
+PYTHON_COMPAT=( python2_7 python3_{6,7,8} )
+
+WEBAPP_MANUAL_SLOT=yes
+WEBAPP_OPTIONAL=yes
+
+# NOTE: Similarly, we cannot go cmake-utils -> cmake until we're on EAPI 7
+inherit cmake-utils depend.apache eapi7-ver perl-functions php-ext-source-r3 python-r1 webapp
+
+DESCRIPTION="Development environment for building spatially enabled webapps"
+HOMEPAGE="https://mapserver.org/"
+SRC_URI="https://download.osgeo.org/mapserver/${P}.tar.gz"
+
+LICENSE="Boost-1.0 BSD BSD-2 ISC MIT tcltk"
+KEYWORDS="~amd64 ~x86"
+SLOT="0"
+
+# NOTE: opengl removed for now as no support for it in upstream CMake
+IUSE="apache bidi cairo geos mysql oracle perl php postgis python"
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+# Tests:
+# Included tests (tests/*) are seriously outdated
+# Upstream's main test suite (msautotest/*) is not in the release tarball,
+# and upstream sets 'export-ignore' for that directory.
+#
+# The eclasses used normally try to run test suites themselves,
+# or skip if nothing was found.
+# However, because of the php-ext-* eclass usage, this fails and would
+# cause errors running non-existent tests, so we have to restrict here.
+RESTRICT="test"
+
+RDEPEND="
+	>=dev-libs/expat-2.2.8
+	dev-libs/libxml2:2=
+	dev-libs/libxslt[crypt]
+	>=dev-libs/protobuf-c-1.3.2:=
+	>=media-libs/freetype-2.9.1-r3
+	>=media-libs/gd-2.0.12:=[truetype,jpeg,png,zlib]
+	>=media-libs/giflib-5.2.1:=
+	>=media-libs/libpng-1.6.37:=
+	>=net-misc/curl-7.69.1
+	>=sci-libs/gdal-3.0.4:=[oracle?]
+	>=sci-libs/proj-6.2.1:=
+	virtual/jpeg
+	virtual/libiconv
+	>=x11-libs/agg-2.5-r3
+	apache? (
+		app-admin/webapp-config
+		dev-libs/fcgi
+	)
+	bidi? (
+		dev-libs/fribidi
+		media-libs/harfbuzz:=
+	)
+	cairo? ( x11-libs/cairo )
+	geos? ( sci-libs/geos )
+	mysql? ( dev-db/mysql-connector-c:= )
+	oracle? ( dev-db/oracle-instantclient:=	)
+	perl? ( dev-lang/perl:= )
+	postgis? (
+		dev-db/postgis
+		dev-db/postgresql:=
+	)
+	python? ( ${PYTHON_DEPS} )
+"
+
+DEPEND="
+	${RDEPEND}
+	virtual/pkgconfig
+	perl? ( >=dev-lang/swig-4.0 )
+	php? ( >=dev-lang/swig-4.0 )
+	python? ( >=dev-lang/swig-4.0 )
+"
```

Comment:
I see `find_python_module(setuptools REQUIRED)` in [mapscript/python/CMakeLists.txt](https://github.com/mapserver/mapserver/blob/rel-7-6-0/mapscript/python/CMakeLists.txt#L4) and I see it in the tagdiff too:
```diff
diff --git a/mapscript/python/CMakeLists.txt b/mapscript/python/CMakeLists.txt
index a9cd007d..678856ce 100644
--- a/mapscript/python/CMakeLists.txt
+++ b/mapscript/python/CMakeLists.txt
@@ -1,14 +1,16 @@
-FIND_PACKAGE(SWIG REQUIRED)
-INCLUDE(${SWIG_USE_FILE})
-FIND_PACKAGE(PythonInterp)
-
+find_package(SWIG REQUIRED)
+include(${SWIG_USE_FILE})
+find_package(PythonInterp)
+find_python_module(setuptools REQUIRED)
```

Do we need to add setuptools as a dependency?

Reply:
You're right that we do -- I'd misremembered setuptools being an implicit dependency now, but that's with the usage of the distutils-* eclass. I've added it now, thanks.